syntax = "proto3";
package tts;

service TTS {
  // client sends one request (or a few), server streams audio frames back
  rpc Synthesize (stream SynthesisRequest) returns (stream AudioChunk);
}

message SynthesisRequest {
  string text = 1;           // text to speak
  int32  sample_rate = 2;    // e.g., 22050
}

message AudioChunk {
  bytes pcm16le = 1;         // raw PCM 16-bit, mono
  int32 sample_rate = 2;     // repeated for convenience
  bool  end_of_stream = 3;
}
